### PR TITLE
Supported board 8ulp 93

### PR DESCRIPTION
**Pull request by angolini**
## Readiness

* [x] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview
This PR includes the flashing procedure for two new boards: **imx93-11x11-lpddr4x-evk** and **imx8ulp-lpddr4-evk**

It also includes a new way to create the board pages using variables. The commits are now separated as I want to ensure this approach makes sense before squashing them.

The reasons I propose this approach are:
* It's easier to copy the board page and know what information needs to be replaced
* It avoids human error while replacing the information (the variable is placed in the right place, so the value will always be equal along the text)

During my tests, it looks like using the same variable name for both machines works fine.

If this approach is accepted, I can:
* [ ] go back to the other boards and change them
* [ ] include other variables needed (such as the tty info)

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 9b7e37590fe083bcdf292b30675e7b6e20519eca are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2061/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 24bcaa9b586670103f480789d9b3562f4b75a62d are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2063/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for d4a5980dceb44b74acd5f108e184f9f3e51c44ee are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2064/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 3778c7237e314fa1c12629081244e704e3b0d18f are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2065/docs/artifacts/html/index.html

File: source/reference-manual/boards/imx8ulpevk.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
+.. _ref-rm_board_imx8ulp-lddr4-evk:
+
+i.MX 8 ULP Evaluation Kit (EVK)
```
Comment (ricardosalveti) [inline source/reference-manual/boards/imx8ulpevk.rst:3]: I noticed we don't have "(EVK)" on every evaluation kits, we should probably align here.

File: source/reference-manual/boards/imx8ulpevk.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+     with ``if02`` in the name (in this example ttyUSB2) and apply the
+     following configuration
+
+          - Baud rate: 115200
+          - Data bits: 8
+          - Stop bit: 1
+          - Parity: None
+          - Flow control: None
+
+#. #. Ensure that the power is off (|power_switch|)
```
Comment (ricardosalveti) [inline source/reference-manual/boards/imx8ulpevk.rst:51]: Extra #. here (renders as 2. 1. ...)

**Review by ricardosalveti — APPROVED** [review]
Minor comments from my side, otherwise lgtm.

**Review by Tim-Anderson — COMMENTED** [review]
I noticed at the end of the flashing instructions it says `put the board into run mode` but does not have the switch settings to change.

File: source/reference-manual/boards/imx8ulpevk.rst (hunk 1 above)
Comment (kprosise) [reply source/reference-manual/boards/imx8ulpevk.rst:3]: @angolini my suggestion is to remove it, at least in the header.

Comment (kprosise) [inline source/reference-manual/boards/imx8ulpevk.rst:3]:
```suggestion
i.MX 8 ULP Evaluation Kit
```

File: source/reference-manual/boards/imx8ulpevk.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
+.. _ref-rm_board_imx8ulp-lddr4-evk:
+
+i.MX 8 ULP Evaluation Kit (EVK)
+===============================
```
Comment (kprosise) [inline source/reference-manual/boards/imx8ulpevk.rst:4]:
```suggestion
=========================
```

File: source/reference-manual/boards/imx8ulpevk.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+Set up the board for updating using the manufacturing tools:
+
+.. figure:: /_static/boards/imx8ulp-lpddr4-evk.png
+     :width: 600
+     :align: center
+
+     |board_name|
+
+#. **OPTIONAL** - Only required if you have problems and/or want to see the boot console output.
```
Comment (kprosise) [inline source/reference-manual/boards/imx8ulpevk.rst:28]:
```suggestion
#. **OPTIONAL**—Only required if you have problems and/or want to see the boot console output.
```

File: source/reference-manual/boards/imx8ulpevk.rst (hunk 2 above)
Comment (kprosise) [inline source/reference-manual/boards/imx8ulpevk.rst:51]:
```suggestion
#. Ensure that the power is off (|power_switch|)
```

File: source/reference-manual/boards/imx8ulpevk.rst [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+     terminal. Three UART connections will appear on the PC.
+     On a Linux host for example::
+
+        $ ls -l /dev/serial/by-id/
+        total 0
+        lrwxrwxrwx 1 root root 13 Feb 13 12:59 usb-0403_FT4232H_9266A2-if00-port0 -> ../../ttyUSB0
+        lrwxrwxrwx 1 root root 13 Feb 13 12:59 usb-0403_FT4232H_9266A2-if02-port0 -> ../../ttyUSB2
+        lrwxrwxrwx 1 root root 13 Feb 13 12:59 usb-0403_FT4232H_9266A2-if03-port0 -> ../../ttyUSB3
+
+     Using a serial terminal program like minicom, connect to the port
```
Comment (kprosise) [inline source/reference-manual/boards/imx8ulpevk.rst:41]:
```suggestion
     Using a serial terminal program like `minicom <https://salsa.debian.org/minicom-team/minicom>`_, connect to the port
```

File: source/reference-manual/boards/imx8ulpevk.rst [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+     with ``if02`` in the name (in this example ttyUSB2) and apply the
+     following configuration
```
Comment (kprosise) [inline source/reference-manual/boards/imx8ulpevk.rst:43]:
```suggestion
     following configuration:
```

File: source/reference-manual/boards/imx8ulpevk.rst [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+#. Connect your computer to the |board_name| board via the USB Type-C port 1 ``Download`` |download_port| jack.
+#. Connect the USB Type-C power plug to the port 2 ``Power`` |power_jack| jack.
+
+#. Power on the |board_name| board by sliding power switch |power_switch| to ON.
+
+Flashing
+--------
+
+Once in serial downloader mode and connected to your PC the evaluation board should show up as an NXP USB device.
```
Comment (kprosise) [inline source/reference-manual/boards/imx8ulpevk.rst:71]:
```suggestion
Once in serial downloader mode and connected to your PC the evaluation board should show up as an NXP® USB device.
```

File: source/reference-manual/boards/imx9-flashing.rst [hunk 8]
```diff
@@ -0,0 +1,50 @@
+.. tabs::
+
+   .. group-tab:: Linux
+
+      1. Verify target is present::
```
Comment (kprosise) [inline source/reference-manual/boards/imx9-flashing.rst:5]:
```suggestion
      #. Verify target is present::
```

File: source/reference-manual/boards/imx93evk.rst [hunk 9]
```diff
@@ -0,0 +1,79 @@
+.. _ref-rm_board_imx93evk:
+
+i.MX 93 Evaluation Kit (EVK)
+============================
```
Comment (kprosise) [inline source/reference-manual/boards/imx93evk.rst:4]:
```suggestion
i.MX 93 Evaluation Kit
=================
```

File: source/reference-manual/boards/imx93evk.rst [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+Set up the board for updating using the manufacturing tools:
+
+.. figure:: /_static/boards/imx93evk.png
+     :width: 600
+     :align: center
+
+     |board_name|
+
+#. **OPTIONAL** - Only required if you have problems and/or want to see the boot console output.
```
Comment (kprosise) [inline source/reference-manual/boards/imx93evk.rst:29]:
```suggestion
#. **OPTIONAL**—Only required if you have problems and/or want to see the boot console output.
```

File: source/reference-manual/boards/imx93evk.rst [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+     On a Linux host for example::
+
+          $ ls -l /dev/serial/by-id/
+          total 0
+          lrwxrwxrwx 1 root root 13 abr 20 16:20 usb-FTDI_Quad_RS232-HS-if00-port0 -> ../../ttyUSB0
+          lrwxrwxrwx 1 root root 13 abr 20 16:20 usb-FTDI_Quad_RS232-HS-if01-port0 -> ../../ttyUSB1
+          lrwxrwxrwx 1 root root 13 abr 20 16:20 usb-FTDI_Quad_RS232-HS-if02-port0 -> ../../ttyUSB2
+          lrwxrwxrwx 1 root root 13 abr 20 16:20 usb-FTDI_Quad_RS232-HS-if03-port0 -> ../../ttyUSB3
+
+     Using a serial terminal program like minicom, connect to the port
```
Comment (kprosise) [inline source/reference-manual/boards/imx93evk.rst:43]:
```suggestion
     Using a serial terminal program like `minicom <https://salsa.debian.org/minicom-team/minicom>`_, connect to the port
```

File: source/reference-manual/boards/imx93evk.rst [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+
+#. Connect your computer to the |board_name| board via the USB Type-C port 1 ``Download`` |download_port| jack.
+#. Connect the USB Type-C power plug to the port 2 ``Power`` |power_jack| jack.
+
+#. Power on the |board_name| board by sliding power switch |power_switch| to ON.
+
+Flashing
+--------
+
+Once in serial downloader mode and connected to your PC the evaluation board should show up as an NXP USB device.
```
Comment (kprosise) [inline source/reference-manual/boards/imx93evk.rst:73]:
```suggestion
Once in serial downloader mode and connected to your PC the evaluation board should show up as an NXP® USB device.
```

**Review by kprosise — COMMENTED** [review]
Made some suggestions!

**Comment by angolini** [comment]
> I noticed at the end of the flashing instructions it says `put the board into run mode` but does not have the switch settings to change.

I will change this for these two boards now, and will change the other board flashing pages after the release

**Comment by angolini** [comment]
accepted all the suggestions, squashed, I think it's ready to merge (if anything else missing)

**Pull request merged**